lib/figo.js

```diff
@@ -181,7 +192,58 @@ var HttpsAgent = function() {
   };
 
   return agent;
-}
+};
+
+
+// constant for number of attempts allowed to make to API, 1 initial and 2 retries, 3 total
+var ALLOWED_NUMBER_OF_ATTEMPTS = 3;
+
+// ### Initialization of https request with a retry count of three.
+//
+// Constructor parameters:
+//
+// - **agent** (`HttpsAgent`) - `HttpsAgent` object
+//
+// - **authorization** (`String`) - encoded authorization in https header
+//
+// - **contentType** (`String`) - content-type for https header
+//
+// - **path** (`String`) - the URL path on the server
+//
+// - **data** (`Object`) - If this parameter is defined, then it will be used as JSON-encoded POST content.
+//
+// - **method** (`Object`) - HTTP method (GET/POST/PUT/DELETE/etc.).
+//
+// - **stringify** (`Object`) - stringify function (either querystring.stringify or JSON.stringify).
+//
+// - **callback** (`Function`) - callback function with two parameters: `error` and `result`
+//
+var queryWithRetries = function (agent, authorization, contentType, contentLength, path, data, method, stringify, callback) {
```

Comment:
- agent is a local variable to the object Session/Connect. It cannot be hard coded in this generalized function
- `Connection`'s `query_api`'s content type is `application/x-www-form-urlencoded` which requires a different type of stringifying `querystring.stringify`, you can read more about this [here](https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options). `Session`'s `query_api`'s content type is `application/json` which requires the stringying function `JSON.stringify`. 
- contentLength, see new commit